reorder_python_imports/main.py

```diff
@@ -247,12 +254,30 @@ def apply_import_sorting(partitions, separate_relative=False, **sort_kwargs):
     new_imports = []
     relative_imports = []
 
+    def _is_require_extra_new_line(last_import_obj, import_obj):
```

Comment:
I'd call this `_import_type_switches`

you could also avoid the `isinstance` calls with something like this:

```
return (
    separate_from_import and 
    last_import_obj is not None and
    type(last_import_obj) is not type(import_obj)
)
```

since there are only two types of imports